cpp/src/parquet/column_reader.h

```diff
@@ -55,6 +56,29 @@ static constexpr uint32_t kDefaultMaxPageHeaderSize = 16 * 1024 * 1024;
 // 16 KB is the default expected page header size
 static constexpr uint32_t kDefaultPageHeaderSize = 16 * 1024;
 
+// \brief DataPageStats stores encoded statistics and number of values/rows for
+// a page.
+struct PARQUET_EXPORT DataPageStats {
+  DataPageStats(EncodedStatistics* encoded_statistics, int32_t num_values,
+                std::optional<int32_t> num_rows)
+      : encoded_statistics(encoded_statistics),
+        is_stats_set(encoded_statistics->is_set()),
+        num_values(num_values),
+        num_rows(num_rows) {}
+
+  // Encoded statistics extracted from the page header.
+  EncodedStatistics* encoded_statistics;
```

Comment:
lets clarify when this might be null?

Reply:
Also, this should perhaps be `const`?

Reply:
I made it a const.

@emkornfield, I made it clear in the comments for setting the callback function that the caller is responsible to check that statistics are correct.